src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const type of action.keysToRemove) {
+        delete state[type];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const del in state) {
+        delete state[del];
+      }
```

Comment:
It's better to use switch/case statement here as you rely on a single value of action.type.